nri-prelude/src/Test/Internal.hs

```diff
@@ -468,13 +470,22 @@ data Summary = Summary
   }
 
 handleUnexpectedErrors :: Expectation -> Expectation
-handleUnexpectedErrors (Expectation task') =
+handleUnexpectedErrors (Expectation task') = do
+  timeout <- timeoutFromEnvOrDefault 10_000
   task'
     |> onException (Task.fail << ThrewException)
-    |> Task.timeout 10_000 TookTooLong
+    |> Task.timeout timeout TookTooLong
     |> Task.onError Task.fail
     |> Expectation
 
+timeoutFromEnvOrDefault :: Prelude.Double -> Expectation' Prelude.Double
+timeoutFromEnvOrDefault defaultTimeout = do
+  let fromIO io = Expectation <| Platform.Internal.Task (\_log -> map Ok io)
```

Comment:
let's move `Expect.fromIO` to be in `Test.Internal` and then re-export it from `Expect` since it imports `Test.Internal` anyway 😎